microcluster/app.go

```diff
@@ -80,9 +84,13 @@ func (m *MicroCluster) Start(ctx context.Context, extensionsSchema []schema.Upda
 		return err
 	}
 
+	if m.args.Version == "" {
```

Comment:
Can we run this check already on `microcluster.App()` where we hand over the args? Then the error message would be much clearer as you could just say the version is missing in the args which were set as part of calling `App()`.

Reply:
I implemented it that way initially; however, we need to initialize `microcluster.App` every time a user wants to make a client request. Since we're only using the version value in the daemon, I didn't want to force clients to supply one if it wasn't going to be used.
I'll clarify the error message; let me know if you'd still prefer that it always be required.

Reply:
I think what we need is two sets of `microcluster.Args`. Right now a lot of args that are daemon-start specific are conflated with general args like `Proxy`, `Client`, and `StateDir`. 


I think it's fine for this PR but I think we should have a follow-up PR that creates a `microcluster.DaemonArgs` that contains all of the arguments that we specify to `microcluster.Start`. 

In fact, maybe this can be an internally defined type in the `daemon package`:

```go
// in the daemon package
type Args struct

// in the microcluster package
type DaemonArgs = daemon.Args
```

And that way the signature for both `daemon.Run` and `microcluster.Start` can be changed to the following:
```go
func (d *Daemon) Run(ctx context.Context, args daemon.Args)

func (m *MicroCluster) Start(ctx context.Context, daemonArgs microcluster.DaemonArgs)
```